client/client.go

```diff
@@ -250,6 +251,27 @@ func parseCertificate(b64 string) (*x509.Certificate, error) {
 	return x509.ParseCertificate(certDER)
 }
 
+// GetProofByHash performs a get-proof-by-hash request, with parameters hash and
+// treeSize.
+// Returned is:
+//   - a GetProofByHashResponse struct, if no error is returned.
+//   - the HTTPData struct returned by GetAndParse() (see above).
+//   - an error, which could be any of the error types returned by
+//     GetAndParse().
+func (lc *LogClient) GetProofByHash(hash []byte, treeSize uint64) (*ct.GetProofByHashResponse, *HTTPData, error) {
+	params := map[string]string{
+		"hash":      base64.StdEncoding.EncodeToString(hash),
```

Comment:
This should be using `base64.URLEncoding`, I think? That's an example of why I think it's better to avoid passing encoded things as much as possible, and keep them raw, so the right encoding can be used...

I *think* this might work nonetheless, because the client might URL-encode parameters so they don't get mangled, but that's not ideal.

Reply:
Doh!  Yup, done :)